### DISCUSSION TITLE
Making multi-select options with query result.

### DISCUSSION THREAD

**Discussion by Jefflai0315** [Budibase/budibase#4519] [Help] [answered]
![image](https://user-images.githubusercontent.com/79297365/154199593-0d82d641-6a8f-46b0-a9ad-2b4ff6343d7a.png)
I am able to display the query result. However, would like to know if I am able to make the item within the result array into the choices of a multi-select option filed?

**Comment by mjashanks** [comment, marked as answer]
Hi @Jefflai0315 

You need to surround your multi-select in a `Data Provider`

Your structure will look like

```
> Form
  > Data provider : Data = Your query
     > Options picker : Source = data provider
```

You'll also need to configure your options picker with `Label` and `value` columns, so that it knows which field to use for it's display.

**Reply by Jefflai0315** [reply]
Thanks for the quick response @mjashanks ! 

![image](https://user-images.githubusercontent.com/79297365/154260553-54e420f3-178c-4ccc-a055-d8b23240c5ad.png)


Instead of having it as a single option, how can I split it into multiple options by comma?

**Reply by mjashanks** [reply]
If you are using a REST query or a custom SQL query, you should use a transformer to do this.

So.
1. Find your query in the `Data` section
2. In the `transformer` box, you should be writing something like 
```javascript
return data.yourFieldName.split(',');
``` 

i.e. `split` will take a comma separated string, and return an array of items.
